app/src/main/java/com/pajato/android/gamechat/chat/fragment/SelectForInviteFragment.java

```diff
@@ -16,47 +16,257 @@
+                break;
+            case R.id.selectorCheck:
+                processSelection(event, (CheckBox) event.view);
+                break;
+            default:
+                break;
+        }
+    }
+
+    /** Process a menu click event ... */
+    @Subscribe public void onClick(final TagClickEvent event) {
+        Object payload = event.view.getTag();
+        if (payload == null || !(payload instanceof MenuEntry)) return;
+
+        // The event represents a menu entry.  Close the FAM and case on the title id.
+        FabManager.chat.dismissMenu(this);
+        MenuEntry entry = (MenuEntry) payload;
+        switch (entry.titleResId) {
+            case R.string.SelectAllMenuTitle:
+                updateSelections(true);
+                break;
+            case R.string.ClearSelectionsMenuTitle:
+                updateSelections(false);
+                break;
+            default:
+                break;
+
+        }
+    }
 
     @Override public void onStart() {
         // Establish the create type, the list type, setup the toolbar and turn off the access
         // control.
         super.onStart();
         ToolbarManager.instance.init(this);
-//        FabManager.chat.setMenu(CHAT_GROUP_FAM_KEY, getGroupMenu());
-
-
-
+        FabManager.chat.setMenu(INVITE_SELECTION_FAM_KEY, getSelectionMenu());
     }
 
     /** Deal with the fragment's lifecycle by managing the progress bar and the FAB. */
     @Override public void onResume() {
         // Set the titles in the toolbar to the app title only; ensure that the FAB is visible, the
-        // FAM is not and the FAM is set to the home chat menu; initialize the ad view; and set up
-        // the group list display.
+        // FAM is not and the FAM is set to the home chat menu.
         super.onResume();
-//        FabManager.chat.setImage(R.drawable.ic_add_white_24dp);
-//        FabManager.chat.init(this, CHAT_GROUP_FAM_KEY);
-//        FabManager.chat.setVisibility(this, View.VISIBLE);
+        FabManager.chat.setImage(R.drawable.ic_add_white_24dp);
+        FabManager.chat.init(this);
+        FabManager.chat.setVisibility(this, View.VISIBLE);
+
+        // Clear selections - is this the correct place to do this (onResume)??????
+        mSelectedGroups.clear();
+        mSelectedRooms.clear();
+    }
+
+    // Private instance methods
+
+    /** Return the home FAM used in the top level show games and show no games fragments. */
+    private List<MenuEntry> getSelectionMenu() {
+        final List<MenuEntry> menu = new ArrayList<>();
+        menu.add(getTintEntry(R.string.SelectAllMenuTitle, R.drawable.ic_done_all_black_24dp));
+        menu.add(getTintEntry(R.string.ClearSelectionsMenuTitle, R.drawable.ic_clear_black_24dp));
+        return menu;
     }
 
+    /** Process a selection */
+    private void processSelection(@NonNull final ClickEvent event, @NonNull final CheckBox checkBox) {
```

Comment:
This method is not working for me.  It needs to be broken up into grokable pieces.